File: LeetCode Easy/121.Best Time To Buy and Sell Stock/sol.java
public class Solution {
    public int maxProfit(int[] prices) {
        int profit = 0;
        int mini = Integer.MAX_VALUE;
        
        for (int price : prices) {
            mini = Math.min(mini, price);
            profit = Math.max(profit, price - mini);
        }
        
        return profit;
    }
}

File: LeetCode Easy/121.Best Time To Buy and Sell Stock/sol.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit=0;
        int mini=INT_MAX;
        for(auto item: prices){
            mini = min(mini,prices[item]);
            profit = max(profit,prices[item]-mini);
        }
        return profit;
    }
};
